src/main/webapp/app/shared/util/firebase/firebase-history-utils.ts

```diff
@@ -335,3 +339,169 @@ export function makeHistoryLocationReadable(locationParts: string[], drugList: r
+  operation: HistoryOperationType | '';
+  timeStamp: number;
+  hugoSymbol: string;
+  old?: string;
+  new?: string;
+  level?: string; // only for treatment add or delete
+};
+
+export type DownloadableHistoryResult = {
+  gene: DownloadableHistoryEntry[];
+  alteration: DownloadableHistoryEntry[];
+  evidence: DownloadableHistoryEntry[];
+};
+
+export function getHistoryEntryStrings(entries: DownloadableHistoryEntry[]) {
+  const sortedEntries = _.sortBy(entries, 'hugoSymbol', 'timeStamp');
+
+  const strings: string[] = [];
+  for (const entry of sortedEntries) {
+    if (!entry.new && !entry.old) {
+      strings.push(
+        `${entry.hugoSymbol} ${entry.location}${entry.level ? ' ' + entry.level : ''} ${HISTORY_OPERATION_TO_PAST_TENSE[entry.operation]}`
+      );
+    } else if ((entry.new || entry.old) && entry.new !== entry.old) {
+      const operationLine = `${entry.hugoSymbol} ${entry.location} ${HISTORY_OPERATION_TO_PAST_TENSE[entry.operation]}\n`;
+      const newLine = `\t New: ${entry.new || ''}\n`;
+      const oldLine = `\t Old: ${entry.old || ''}`;
+
+      strings.push(operationLine + newLine + oldLine);
+    }
+  }
+  return _.uniq(strings);
+}
+
+export function getAllGeneHistoryForDateRange(historyCollection: HistoryCollection, drugList: readonly IDrug[], start?: Date, end?: Date) {
+  const result: DownloadableHistoryResult = {
+    gene: [],
+    alteration: [],
+    evidence: [],
+  };
+
+  if (end && end < start) {
+    return result;
+  }
+
+  for (const [hugoSymbol, history] of Object.entries(historyCollection)) {
+    const geneHistory = getGeneHistoryForDateRange(hugoSymbol, history.api, drugList, start, end);
+    result.gene.push(...geneHistory.gene);
+    result.alteration.push(...geneHistory.alteration);
+    result.evidence.push(...geneHistory.evidence);
+  }
+
+  return result;
+}
+
+export function isBetweenDates(date: Date, start?: Date, end?: Date) {
+  if (start && end) {
+    return date >= start && date <= end;
+  } else if (start) {
+    return date >= start;
+  } else if (end) {
+    return date <= end;
+  } else {
+    return true;
+  }
+}
+
+export function getGeneHistoryForDateRange(
+  hugoSymbol: string,
+  historyList: HistoryList,
+  drugList: readonly IDrug[],
+  start?: Date,
+  end?: Date
+) {
+  const result: DownloadableHistoryResult = {
+    gene: [],
+    alteration: [],
+    evidence: [],
+  };
+
+  if (end && end < start) {
+    return result;
+  }
+
+  const geneFields = [
+    READABLE_FIELD.GENE_TYPE,
+    READABLE_FIELD.SUMMARY,
+    READABLE_FIELD.BACKGROUND,
+    READABLE_FIELD.PENETRANCE,
+    READABLE_FIELD.INHERITANCE_MECHANISM,
+  ];
+  const alterationFields = [
+    READABLE_FIELD.MUTATION_EFFECT,
+    READABLE_FIELD.MUTATION_SPECIFIC_INHERITANCE,
+    READABLE_FIELD.MUTATION_SPECIFIC_PENETRANCE,
+  ];
+  const showDiffFields = [READABLE_FIELD.ONCOGENIC, READABLE_FIELD.EFFECT, READABLE_FIELD.LEVEL];
+
+  const filteredHistory = _.cloneDeep(historyList);
```

Comment:
Is this a big object? (Asking in case there's a performance issue)

Reply:
Yeah, it's a decent sized file. BRAF is about 1mb for reference. I was a little concerned with cloning here too, but the operation feels pretty instantaneous

Reply:
sounds good we can see how it goes